tests/connect_redirect/connect_redirect_tests.cpp

```diff
@@ -0,0 +1,482 @@
+    printf("AUTH: Local | %s | %s | %s\n", protocol_string, family_string, dual_stack_string);
+    authorize_test_wrapper(object, dual_stack, addresses.local_address);
+
+    // Second category is connection redirection tests.
+
+    // Remote -> Remote
+    printf("REDIRECT: Remote -> Remote | %s | %s | %s\n", protocol_string, family_string, dual_stack_string);
+    connect_redirect_test_wrapper(object, addresses.vip_address, addresses.remote_address, dual_stack);
+
+    // Remote -> Loopback
+    printf("REDIRECT: Remote -> Loopback | %s | %s | %s\n", protocol_string, family_string, dual_stack_string);
+    connect_redirect_test_wrapper(object, addresses.vip_address, addresses.loopback_address, dual_stack);
+
+    // Remote -> Local non-loopback
+    printf("REDIRECT: Remote -> Local | %s | %s | %s\n", protocol_string, family_string, dual_stack_string);
+    connect_redirect_test_wrapper(object, addresses.vip_address, addresses.local_address, dual_stack);
+
+    // Loopback -> Remote
+    printf("REDIRECT: Loopback -> Remote | %s | %s | %s\n", protocol_string, family_string, dual_stack_string);
+    connect_redirect_test_wrapper(object, addresses.loopback_address, addresses.remote_address, dual_stack);
+
+    // Loopback -> Local non-loopback
+    printf("REDIRECT: Loopback -> Local | %s | %s | %s\n", protocol_string, family_string, dual_stack_string);
+    connect_redirect_test_wrapper(object, addresses.loopback_address, addresses.local_address, dual_stack);
+
+    // Local non-loopback -> Loopback
+    printf("REDIRECT: Local -> Loopback | %s | %s | %s\n", protocol_string, family_string, dual_stack_string);
+    connect_redirect_test_wrapper(object, addresses.local_address, addresses.loopback_address, dual_stack);
+
+    // Local non-loopback -> Remote
+    printf("REDIRECT: Local -> Remote | %s | %s | %s\n", protocol_string, family_string, dual_stack_string);
+    connect_redirect_test_wrapper(object, addresses.local_address, addresses.remote_address, dual_stack);
+}
+
+TEST_CASE("connect_redirect_tcp_v4", "[connect_redirect_tests]")
+{
+    _initialize_test_globals();
+
+    struct bpf_object* object = nullptr;
+    _load_and_attach_ebpf_programs(&object);
+
+    // Test for IPv4 traffic.
+    _globals.family = AF_INET;
+
+    _globals.protocol = IPPROTO_TCP;
+    connect_redirect_tests_common(object, false, _globals.addresses[socket_family_t::IPv4]);
+    connect_redirect_tests_common(object, true, _globals.addresses[socket_family_t::Dual]);
+
+    // This should also detach the programs as they are not pinned.
+    bpf_object__close(object);
+}
+
+TEST_CASE("connect_redirect_tcp_v6", "[connect_redirect_tests]")
+{
+    _initialize_test_globals();
+
+    struct bpf_object* object = nullptr;
+    _load_and_attach_ebpf_programs(&object);
+
+    // Test for IPv6 traffic.
+    _globals.family = AF_INET6;
+
+    _globals.protocol = IPPROTO_TCP;
+    connect_redirect_tests_common(object, false, _globals.addresses[socket_family_t::IPv6]);
```

Comment:
same comment; for readability please dont pass literal true/false. At least put a comment with parameter name.

Reply:
added comment